src/Functions/FunctionsHashing.h

```diff
@@ -1240,17 +1310,31 @@ class FunctionAnyHash : public IFunction
 
         typename ColumnVector<ToType>::Container & vec_to = col_to->getData();
 
-        if (arguments.empty())
+        /// If using a "keyed" algorithm, the first argument is the key and
+        /// the data starts from the second argument.
+        /// Otherwise there is no key and all arguments are interpreted as data.
+        constexpr size_t first_data_argument = Keyed;
+
+        if (arguments.size() <= first_data_argument)
         {
             /// Constant random number from /dev/urandom is used as a hash value of empty list of arguments.
             vec_to.assign(rows, static_cast<ToType>(0xe28dbde7fe22e41c));
         }
 
-        /// The function supports arbitrary number of arguments of arbitrary types.
+        KeyType key{};
+        if constexpr (Keyed)
```

Comment:
In general, feel free to omit parentheses for single-line if (constexpr) statements, e.g.
```cpp
if constexpr (Keyed)
    if (!arguments.empty())
        key = Impl::parseKey(arguments[0]);
```